### PR TITLE
Refactor authentication

### PR DESCRIPTION
**Pull request by michaelsena**
This PR:

- changed the structure of the authentication section to be DID Method centric
- added detail to various pages within each DID Method
- Streamlined the authentication page in the build section

### CODE REVIEW

File: docs/authentication/3id-did/method.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+## **Considerations**
+
+**Mutable DID document**: The 3ID DID Method is fully implemented on top of Ceramic. The primary benefit of the 3ID DID method is that it has a **mutable DID document**. The DID Document for 3ID DIDs is stored in a [TileDocument StreamType](../../streamtypes/tile-document/overview.md). This is enables 3ID DIDs to handle multiple keys, secure key rotations, and the addition and removal or arbitrary data.
+
+**Aggregated identities**: 3ID DIDs can serve as a cross-chain identifier which can be controlled by all of a user's blockchain and Web3 accounts from any L1 or L2 protocol. This provides a way to unify a user's identity across all other platforms.
+
+**Seamless integration with blockchain wallets**: When 3ID DIDs are used in conjunction with [IDX (CIP-11)](../../tools/identity/idx.md) (using the [3ID Keychain definition (CIP-20)](https://github.com/ceramicnetwork/CIP/blob/main/CIPs/CIP-20/CIP-20.md)), a 3ID DID can be controlled from a user's existing blockchain wallets. This functionality is implemented by [3ID Connect](./3id-connect.md).
```
Comment (stbrody) [inline docs/authentication/3id-did/method.md:11]: Is this right?  Why would using IDX be necessary to use 3IDs with blockchain wallets?

File: docs/authentication/3id-did/method.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
+# 3ID DID Method
+
+The [3ID DID Method (CIP-79)](https://github.com/ceramicnetwork/CIP/blob/main/CIPs/CIP-79/CIP-79.md) is a [DID method](../../learn/glossary.md#did-methods) which can be used to [authenticate](../../build/authentication.md) to Ceramic and perform [writes](../../build/writes.md) to streams that rely on DIDs. 3ID DID is a powerful DID method that supports multiple keys, key rotations, and revocations. The 3ID DID Method is on the W3C's official DID method registry and is fully compliant with decentralized identity standards.
+
+## **Considerations**
+
+**Mutable DID document**: The 3ID DID Method is fully implemented on top of Ceramic. The primary benefit of the 3ID DID method is that it has a **mutable DID document**. The DID Document for 3ID DIDs is stored in a [TileDocument StreamType](../../streamtypes/tile-document/overview.md). This is enables 3ID DIDs to handle multiple keys, secure key rotations, and the addition and removal or arbitrary data.
```
Comment (stbrody) [inline docs/authentication/3id-did/method.md:7]: I feel like secure key rotations is noteworthy enough to be a top-level bullet.  I think that's one of the truly revolutionary things about 3ID over other DID methods

File: docs/authentication/3id-did/method.md [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+Below, find a simplified version of the 3ID DID Method specification. For the full specification, view [3ID DID Method (CIP-79)](https://github.com/ceramicnetwork/CIP/blob/main/CIPs/CIP-79/CIP-79.md).
+
+### Method name
+The official method name for the 3ID DID method is `3`.
+
+### DID string
+Example 3ID DID identifier:
+
+```
+did:3:<method-specific-identifier>
```
Comment (stbrody) [inline docs/authentication/3id-did/method.md:39]:
```suggestion
did:3:<StreamID of the Ceramic TileDocument that contains the DID document>
```

Comment (stbrody) [reply docs/authentication/3id-did/method.md:39]: optional, but might be nice to be more specific here and show how it uses Ceramic.

File: docs/authentication/3id-did/provider.md [hunk 4]
```diff
@@ -0,0 +1,105 @@
+# 3ID DID Provider
```
Comment (stbrody) [inline docs/authentication/3id-did/provider.md:1]: I don't feel well equipped to review this page. I'll defer to @oed here

File: docs/authentication/key-did/method.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
+# Key DID Method
+
+The Key DID Method is a [DID method](../../learn/glossary.md#did-methods) which can be used to [authenticate](../../build/authentication.md) to a Ceramic client to perform [writes]() to streams that rely on DIDs for authentication. The Key DID Method is the most simple DID method. It simply encodes a public key in the DID string, and when resolved converts this public key into a [DID Document](../../learn/glossary.md#did-document). Key DID is on the W3C's official DID method registry and is fully compliant with decentralized identity standards. Carefully read the considerations below before deciding to use the Key DID Method in your project.
```
Comment (stbrody) [inline docs/authentication/key-did/method.md:3]: missing link for writes

File: docs/authentication/key-did/method.md [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+## **Considerations**
+
+**One key only**: The DID Document for a Key DID is explicitly tied to a single crypto key. It can not support multiple keys in the DID document nor can it support key rotation, which means only one key can ever control the DID and it can never be changed in case it is compromised.
+
+**For advanced users**: For the reasons above, the Key DID Method is only suitable for advanced users who will only want to ever use one keypair to control their DID, and who have strong key security practices - such as a developer.
```
Comment (stbrody) [inline docs/authentication/key-did/method.md:9]:
```suggestion
**For advanced users**: For the reasons above, the Key DID Method is only suitable for advanced users who will only want to ever use one keypair to control their DID, and who have strong key security practices - such as a developer - and so generally will not be appropriate for identities for non-technical end users.
```

File: mkdocs.yml [hunk 7]
```diff
@@ -47,16 +46,22 @@ nav:
...
+    - JavaScript:
+      - JS HTTP Client: clients/javascript/http.md
+      - JS Core Client: clients/javascript/core.md
+      - CLI: clients/javascript/cli.md
   - Run:
-    - Running a node: run/nodes.md
+    - Nodes:
+      - Running a node: run/nodes.md
     #- Anchor service: run/anchor-service.md
+    - Anchor service:
```
Comment (stbrody) [inline mkdocs.yml:64]: empty section?

**Comment by michaelsena** [comment]
Thanks @stbrody. @oed I still would like a check from to make sure everything makes sense and that all the code flows are good.

File: docs/authentication/3id-did/method.md (hunk 2 above)
Comment (michaelsena) [reply docs/authentication/3id-did/method.md]: made this change

File: docs/authentication/3id-did/method.md [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+
+## **Considerations**
+
+**Mutable DID document**: The 3ID DID Method is fully implemented on top of Ceramic. The primary benefit of the 3ID DID method is that it has a **mutable DID document**. The DID Document for 3ID DIDs is stored in a [TileDocument StreamType](../../streamtypes/tile-document/overview.md). This is enables 3ID DIDs to handle multiple keys, secure key rotations, and the addition and removal or arbitrary data.
+
+**Aggregated identities**: 3ID DIDs can serve as a cross-chain identifier which can be controlled by all of a user's blockchain and Web3 accounts from any L1 or L2 protocol. This provides a way to unify a user's identity across all other platforms.
```
[the remark this answers is not in the document]
Comment (michaelsena) [reply docs/authentication/3id-did/method.md]: changed the order so this is below the paragraph on integration with blockchain wallets

File: docs/authentication/3id-did/method.md (hunk 1 above)
Comment (michaelsena) [reply docs/authentication/3id-did/method.md:9]: going to leave as is for now, we can revisit

File: docs/authentication/3id-did/method.md (hunk 3 above)
Comment (michaelsena) [reply docs/authentication/3id-did/method.md]: made this fix

File: docs/authentication/3id-did/provider.md [hunk 9]
```diff
@@ -0,0 +1,105 @@
...
+
+### 4. Create a 3ID Provider instance
+
+#### Option 1: Using the seed
+
+``` js
+const threeId = await ThreeIdProvider.create({ getPermission, seed })
+const provider = threeId.getDidProvider()
+
+```
```
[the remark this answers is not in the document]
Comment (michaelsena) [reply docs/authentication/3id-did/provider.md:36]: @oed so you're just saying we omit the option to create from seed? What is people use this or want to, shouldn't it still be documented? Perhaps we replace the ordering of 1 and 2 and we mark the authprovider as recommended.

File: docs/authentication/3id-did/provider.md (hunk 4 above)
Comment (michaelsena) [reply docs/authentication/3id-did/provider.md:1]: @oed have you had a chance to look at the 3id-did-provider page?

File: docs/authentication/key-did/method.md (hunk 5 above)
Comment (michaelsena) [reply docs/authentication/key-did/method.md]: fixed

File: mkdocs.yml (hunk 7 above)
Comment (michaelsena) [reply mkdocs.yml]: fixed

File: docs/authentication/3id-did/3id-connect.md [hunk 10]
```diff
@@ -98,14 +96,14 @@ await threeIdConnect.connect(authProvider)
 const provider = await threeIdConnect.getDidProvider()
 ```
 
-### 6. Set the provider
+### 6. Set the provider to Ceramic
 Set the Provider instance to the DID instance used by your Ceramic client in order to perform writes.
 
 ```
 ceramic.did.setProvider(provider)
```
[the remark this answers is not in the document]
Comment (michaelsena) [reply docs/authentication/3id-did/3id-connect.md:103]: cc @stbrody just this last one remaining...

File: docs/authentication/3id-did/provider.md (hunk 9 above)
[the remark this answers is not in the document]
Comment (michaelsena) [reply docs/authentication/3id-did/provider.md]: I reordered for option 2 to appear first, and be recommended. Also added description to the titles to be more clear it's `Using the 3ID's seed`

File: docs/authentication/3id-did/3id-connect.md (hunk 10 above)
[the remark this answers is not in the document]
Comment (michaelsena) [reply docs/authentication/3id-did/3id-connect.md]: We created the did instance during installation of the client. I added a sentence here (and on other provider pages) that reminds people of this.

**Comment by michaelsena** [comment]
All comments have now been addressed.

**Review by stbrody — APPROVED** [review]

**Pull request merged**